File: Chapter 03/C#/Section_3_3_2_OCP/After/DribblePastOpponent.cs
using System;

namespace Chapter03.OCP.After;

/// <summary>
/// A concrete implementation of the IPlay interface.
/// </summary>
public class DribblePastOpponent : IPlay
{
    public void Execute()
    {
        Console.WriteLine("  [Action] Executing a dribble move…");
    }
}

File: Chapter 03/JS/section_3_3_2_ocp/after/dribblePastOpponent.js
/**
 * A Concrete Strategy.
 * * ARCHITECTURE NOTE: This class is a standalone implementation of a specific play.
 * By keeping it separate, we ensure that changes to the 'Dribble' logic 
 * never impact the 'Pass' logic or the 'Midfielder' core logic.
 */
class DribblePastOpponent {
    execute() {
        console.log("  [Action] Executing a dribble move…");
    }
}

module.exports = DribblePastOpponent;